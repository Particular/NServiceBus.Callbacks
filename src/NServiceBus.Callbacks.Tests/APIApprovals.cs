using NServiceBus;
using NUnit.Framework;
using Particular.Approvals;
using PublicApiGenerator;

[TestFixture]
public class APIApprovals
{
    [Test]
    public void Approve()
    {
        var publicApi = ApiGenerator.GeneratePublicApi(typeof(CallbackExtensions).Assembly, excludeAttributes: new[] { "System.Runtime.Versioning.TargetFrameworkAttribute" });
        Approver.Verify(publicApi);
    }
}