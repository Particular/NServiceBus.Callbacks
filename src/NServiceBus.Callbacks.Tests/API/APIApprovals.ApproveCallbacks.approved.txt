[assembly: System.CLSCompliantAttribute(true)]
[assembly: System.Runtime.InteropServices.ComVisibleAttribute(false)]
[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.5.1", FrameworkDisplayName=".NET Framework 4.5.1")]
namespace NServiceBus
{
    public class LegacyEnumResponse<T>
    {
        public LegacyEnumResponse(T status) { }
        public T Status { get; }
    }
    public class static SendOptionsExtensions
    {
        public static NServiceBus.SendOptions RegisterCancellationToken(this NServiceBus.SendOptions options, System.Threading.CancellationToken cancellationToken) { }
    }
    public class static TransientRequestResponseExtensions
    {
        public static System.Threading.Tasks.Task<TResponse> RequestWithTransientlyHandledResponseAsync<TResponse>(this NServiceBus.IBus bus, object requestMessage, NServiceBus.SendOptions options) { }
    }
}